### PR TITLE
Add performance report for latest version with minor fixes to scripts

### PR DESCRIPTION
**Pull request by glpuga**
### Proposed changes

- Fixes a permissions issue that prevented scripts in `beluga_benchmark` to run if not built with `--symlink-install`
- Adds the `--params-file` optional argument to the `parameterized_run` script, which was cited in reports, but was missing from script.
- Adds an up-to-date report comparing `beluga_amcl` to `nav2_amcl`, here: [link to report](https://github.com/Ekumen-OS/beluga/blob/glpuga/report_may/beluga_benchmark/docs/reports/2023-05-20/REPORT.md)
- Includes a summary of the report in the `beluga_amcl` README.md
- Improves the output of the `compare_results.py` script by adding a grid to the plots and making the axes logarithmic.

#### Type of change

- [x] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [x] 📚 Documentation (change which fixes or extends documentation)

### Checklist

_Put an `x` in the boxes that apply. This is simply a reminder of what we will require before merging your code._

- [x] Lint and unit tests (if any) pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commmits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

**Comment by glpuga** [comment]
@nahueespinosa If we confirm these results, the multi threaded configuration should not be the default for normal amounts of particles. Look at the CPU usage for beluga in both cases.

### Beluga (non-multithreaded) vs. AMCL

beluga_seq_vs_amcl

### Beluga (multithreaded) vs. AMCL

beluga_par_vs_amcl

**Comment by glpuga** [comment]
I extended the report with results for the `beam` model too. The report is here: https://github.com/Ekumen-OS/beluga/blob/glpuga/report_may/beluga_benchmark/docs/reports/2023-05-20/REPORT.md

File: beluga_amcl/README.md [hunk 1]
```diff
@@ -53,6 +52,20 @@ Defaults are `map`, `odom` and `base`.
 |------------------------------------|------------------|-------------------------------------------------------------------------------|
 | `reinitialize_global_localization` | `std_srvs/Empty` | Request to reinitialize global localization without an initial pose estimate. |
 
+## Performance
+
+In the [beluga_benchmark](../beluga_benchmark) package a set of scripts can be found that benchmark the performance `beluga_amcl` against that of `nav2_amcl` using a synthetic dataset.
```
Comment (nahueespinosa) [inline beluga_amcl/README.md:57]:
```suggestion
In the [beluga_benchmark](../beluga_benchmark) package a set of scripts can be found that benchmark the performance of `beluga_amcl` against that of `nav2_amcl` using a synthetic dataset.
```

File: beluga_amcl/README.md [hunk 2]
```diff
@@ -53,6 +52,20 @@ Defaults are `map`, `odom` and `base`.
...
+
+The following plot displays the RSS (Resident Set Size), CPU usage and APE (Absolute Pose Error) statistics for both  `beluga_amcl` and `nav2_amcl`, with particle sizes ranging between 250 and 200000 and sensor model `likelihood field`.
+
+![Beluga vs Nav2 AMCL](../beluga_benchmark/docs/reports/2023-05-20/likelihood_seq_beluga_vs_amcl.png)
+
+The following plot shows the RSS (Resident Set Size), CPU usage and APE (Absolute Pose Error) statistics for both  `beluga_amcl` and `nav2_amcl`, with particle sizes ranging between 250 and 200000 and sensor model `beam`.
```
Comment (nahueespinosa) [inline beluga_amcl/README.md:63]: nit: Just to use the same phrasing as the previous paragraph.
```suggestion
The following plot displays the RSS (Resident Set Size), CPU usage and APE (Absolute Pose Error) statistics for both  `beluga_amcl` and `nav2_amcl`, with particle sizes ranging between 250 and 200000 and sensor model `beam`.
```

File: beluga_benchmark/docs/reports/2023-05-20/REPORT.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+- CPU usage.
+- APE (Absolute Pose Error) statistics: `mean`, `median`, `max` and `rmse`.
+
+## Results
+
+#### Non-multithreaded Beluga vs. Nav2 AMCL with Likelihood Field Sensor Model
+
+![Non-multithreaded Beluga vs. Nav2 AMCL with Likelihood Field Sensor Model](likelihood_seq_beluga_vs_amcl.png)
+
+Comentary on the results:
```
Comment (nahueespinosa) [inline beluga_benchmark/docs/reports/2023-05-20/REPORT.md:43]:
```suggestion
Commentary on the results:
```

File: beluga_benchmark/docs/reports/2023-05-20/REPORT.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+- Memory usage for `beluga_amcl` grows slower than for `nav2_amcl` throughout the whole range of particle counts.
+- `beluga_amcl` has about the same level of CPU usage as `nav2_amcl` under $10^4$ particles. Beyond that point both non-multithreaded implementations saturate near 100% CPU usage.
+- The trajectory APE is on par or slightly better for `beluga_amcl` than `nav2_amcl` throughout the whole range of particles. In the CPU saturation region, the localization performance for `nav2_amcl` suddenly degrades while `beluga_amcl` continues to provide some level of service.
+
+#### Multithreaded Beluga vs. Nav2 AMCL with Likelihood Field Sensor Model
+
+![Multithreaded Beluga vs. Nav2 AMCL with Likelihood Field Sensor Model](likelihood_par_beluga_vs_amcl.png)
+
+Comentary on the results:
```
Comment (nahueespinosa) [inline beluga_benchmark/docs/reports/2023-05-20/REPORT.md:53]:
```suggestion
Commentary on the results:
```

File: beluga_benchmark/docs/reports/2023-05-20/REPORT.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+- The APE is on par for both nodes below $50k$ particles. Beyond this point both nodes degrade, but `nav2_amcl` degrades faster suddenly, while `beluga_amcl` degrades more gradually.
+
+## Conclusions
+
+- `beluga_amcl` provides better memory usage than `nav2_amcl` in all cases.
+- In all configurations, `beluga_amcl` provides similar localization error performance to `nav2_amcl`, degrading more gradually in the CPU saturation region.
+- CPU usage for `beluga_amcl` is higher than for `nav2_amcl` both for:
+  - `beluga_amcl` with the `beam sensor` model.
+  - `beluga_amcl` in multithreaded mode.
+- These experiments do not account for the reduction in iteration latency reduction brought by multithreading. This is a topic for future work.
```
Comment (nahueespinosa) [inline beluga_benchmark/docs/reports/2023-05-20/REPORT.md:82]:
```suggestion
- These experiments do not account for the reduction in iteration latency brought by multithreading. This is a topic for future work.
```

**Review by nahueespinosa — COMMENTED** [review]
@glpuga Awesome!

> the multi threaded configuration should not be the default for normal amounts of particles

I agree, we knew of the overhead associated with multi-threading,

**Review by nahueespinosa — APPROVED** [review]
LGTM!

**Pull request merged**